sdk/identity/azure-identity/azure/identity/_credentials/device_code.py

```diff
@@ -41,12 +42,19 @@ class DeviceCodeCredential(InteractiveCredential):
         - ``expires_on`` (datetime.datetime) the UTC time at which the code will expire
         If this argument isn't provided, the credential will print instructions to stdout.
     :paramtype prompt_callback: Callable[str, str, ~datetime.datetime]
+
     :keyword AuthenticationRecord authentication_record: :class:`AuthenticationRecord` returned by :func:`authenticate`
     :keyword bool disable_automatic_authentication: if True, :func:`get_token` will raise
         :class:`AuthenticationRequiredError` when user interaction is required to acquire a token. Defaults to False.
     :keyword cache_persistence_options: configuration for persistent token caching. If unspecified, the credential
         will cache tokens in memory.
     :paramtype cache_persistence_options: ~azure.identity.TokenCachePersistenceOptions
+    :keyword bool allow_broker: An authentication broker is an application that runs on a user's machine that manages
```

Comment:
WAM does not work for device code.

We only need browser and username/password.